File: client/src/app/models/task.ts
import {User} from '@models/user';
import {TaskComment} from '@models/task-comment';

export class Task {
  public id: number;
  public name: string;
  public status: string;
  public deadline: Date;
  public color: string;
  public priority: string;

  public reminderAt?: Date;
  public description?: string;

  public parent?: Task|null;
  public assignees?: User[];
  public comments? : TaskComment[];
  constructor(
    id: number,
    name: string,
    status: string,
    deadline: Date,
    color: string,
    priority: string,
    description?: string,
    reminderAt?: Date,
    parent?: Task,
    assignees?: User[],
    comments?: TaskComment[]
  ) {
    this.id = id;
    this.name = name;
    this.status = status;
    this.deadline = deadline;
    this.color = color;
    this.priority = priority;``

    this.description = description;

    this.parent = parent;
    this.reminderAt = reminderAt;
    this.assignees = assignees;
    this.comments = comments;
  }
}

File: server/WebApplication1/Models/Task.cs
namespace WebApplication1.Models
{
    public class Task
    {
        public int id { get; set; }
        public string name { get; set; }
        public string? description { get; set; }
        public int? group_id { get; set; }
        public string status { get; set; }
        public DateTime? deadline { get; set; }
        public string? color { get; set; }
        public string? priority { get; set; }
        public string? name_alt { get; set; }
        public int created_by { get; set; }
        public int? updated_by { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }
        public DateTime? deleted_at { get; set; }
        public int? parent_id { get; set; }
        public DateTime? reminder_at { get; set; }


    }
}
